### PR TITLE
Add Elyra to list of notebooks available for ODH

### PR DESCRIPTION
**Pull request by akchinSTC**
- Adds Elyra build configs and Imagestream

Will need to rebase images to official ones in dockerhub when they get published

### CODE REVIEW

File: jupyterhub/jupyterhub/base/jupyterhub-singleuser-profiles-configmap.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ data:
             memory: 2Gi
             cpu: 1
 
+      - name: Elyra Notebook
```
Comment (vpavlin) [inline jupyterhub/jupyterhub/base/jupyterhub-singleuser-profiles-configmap.yaml:20]: We don't need this right now..the image will be listed based on the image streams

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  name: s2i-elyra-notebook
+spec:
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+      openshift.io/imported-from: quay.io/akchinstc/s2i-elyra-notebook
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:12]: Can you please change the URL to `quay.io/thoth-station/s2i-lab-elyra:v0.0.1` (we'll update it once your other PR is merged)

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    from:
+      kind: DockerImage
+      name: quay.io/akchinstc/s2i-elyra-notebook:1.0.0
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:15]: `quay.io/thoth-station/s2i-lab-elyra:v0.0.1`

File: jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  name: s2i-elyra-notebook
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: s2i-elyra-notebook:1.0.0
+  source:
+    contextDir: elyra-notebook
+    git:
+      uri: https://github.com/akchinSTC/jupyter-notebooks
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml:15]: Ca you change the URL to https://github.com/opendatahub-io/s2i-lab-elyra

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/notebook-image: "true"
+  name: s2i-elyra-notebook
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:6]: Can you add similar annotations: https://github.com/opendatahub-io/odh-manifests/blob/master/jupyterhub/notebook-images/overlays/additional/minimal-notebook-imagestream.yaml#L7-L9, please?

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+      openshift.io/imported-from: quay.io/akchinstc/s2i-elyra-notebook
+    from:
+      kind: DockerImage
+      name: quay.io/akchinstc/s2i-elyra-notebook:1.0.0
+    name: "1.0.0"
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:16]: make this `v0.0.1`

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  name: s2i-lab-elyra
+spec:
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+      openshift.io/imported-from: quay.io/thoth-station/s2i-lab-elyra
+    from:
+      kind: DockerImage
+      name: quay.io/thoth-station/s2i-lab-elyra:0.0.1
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:19]: The tag is `v0.0.1` (see https://quay.io/repository/thoth-station/s2i-lab-elyra?tag=v0.0.1&tab=tags)

File: jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+kind: BuildConfig
+metadata:
+  labels:
+    build: s2i-lab-elyra
+  name: s2i-lab-elyra
+spec:
+  output:
+    to:
+      kind: ImageStreamTag
+      name: quay.io/thoth-station/s2i-lab-elyra:0.0.1
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml:11]: This has to match the image stream name and tag, so : `s2i-lab-elyra:v0.0.1`

File: jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  source:
+    git:
+      uri: https://github.com/opendatahub-io/s2i-lab-elyra
+    type: Git
+  strategy:
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: quay.io/thoth-station/s2i-minimal-notebook:v0.0.4
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/build/elyra-notebook-buildconfig.yaml:20]: This should also match the ImageStream name, so `s2i-minimal-notebook:v0.0.4`

File: jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  name: s2i-lab-elyra
+spec:
+  lookupPolicy:
+    local: true
+  tags:
+  - annotations:
+      openshift.io/imported-from: quay.io/thoth-station/s2i-lab-elyra
+    from:
+      kind: DockerImage
+      name: quay.io/thoth-station/s2i-lab-elyra:v0.0.1
```
Comment (vpavlin) [inline jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml:19]: @akchinSTC I have merged your PR in s2i-lab-elyra repo..Would you please make one last change - replace version `v0.0.1` with `v0.0.2` and I think we are ok to merge

**Comment by crobby** [comment]
/test all
ci system continues to flake

**Pull request merged**